.eslintrc

```diff
@@ -60,7 +60,7 @@
         "import/no-duplicates": "error",
         "@typescript-eslint/no-floating-promises": 2,
         "@typescript-eslint/no-unused-vars": 2,
-        "typesafe/no-throw-sync-func": "error"
+        "typesafe/no-throw-sync-func": "off"
```

Comment:
why?

Reply:
@kwajiehao and i had an earlier [discussion](https://github.com/opengovsg/sgid-client/pull/35#discussion_r1177290124) about whether we want to enforce this practice of not throwing errors in (synchronous) function

conclusion was

> after some discussion we've decided that this ESLint rule, while it does have its use, is quite opinionated and not necessarily considered a standard practice (many Python and Node libraries throw errors in synchronous functions) and as such we will be removing it

> additionally, as our other codebases are not adhering to this specific ESLint rule, we've decided not to adhere to it as part of an effort to standardize our coding practices

feel free to respond if you would like to reopen and contribute to the discussion

Reply:
yeah.. I mean just remove it instead of turning off if not required.

Reply:
ahh that makes sense - done! thank you :D